docs: Document limitations.

jodh-intel opened the pull request:
Add a document listing the known limitations.

Fixes #338.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

jodh-intel commented:
Just noticed that the issue links aren't "active", but I think that's because the document isn't yet merged into the master branch.

jodh-intel commented:
@iphutch - could you take a look at this please?

grahamwhaley commented:
lgtm

gorozco1 commented on docs/limitations.md:
```diff
@@ -0,0 +1,296 @@
+through to the QEMU `-m` memory size option. This is also related to the
+`docker update` command.
+
+See issue #381 for more information.
```
missed link to issue to make the document easy to read and to relate issues faster.

gorozco1 commented on docs/limitations.md:
```diff
@@ -0,0 +1,296 @@
+the QEMU command line CPU configuration options to gain a similar
+effect.
+
+See issue #341 for more information.
```
missed link to issue to make the document easy to read and to relate issues faster.

gorozco1 commented on docs/limitations.md:
```diff
@@ -0,0 +1,296 @@
+It should be possible to pass this information through to the QEMU
+command line CPU configuration options to gain a similar effect.
+
+See issue #388 for more information.
```
missed link to issue

gorozco1 commented on docs/limitations.md:
```diff
@@ -0,0 +1,296 @@
+items, these capabilities could be modified either in the host, in the
+VM, or potentially both.
+
+See issue #51 for more information.
```
missed link to issue

gorozco1 commented on docs/limitations.md:
```diff
@@ -0,0 +1,296 @@
+the whole VM container as viewed from the host. The latter is likely
+more useful from a whole system point of view.
+
+See issue #200 for more information.
```
missed link to issue

gorozco1 commented on docs/limitations.md:
```diff
@@ -0,0 +1,296 @@
+
+The Clear Containers runtime does not currently support the `ps` command.
+
+See issue #95 for more information.
```
missed link to issue

gorozco1 commented on docs/limitations.md:
```diff
@@ -0,0 +1,296 @@
+See here for the
+[runc implementation](https://github.com/opencontainers/runc/blob/e775f0fba3ea329b8b766451c892c41a3d49594d/events.go).
+
+See issue #379 for more information.
```
missed link to issue

gorozco1 commented on docs/limitations.md:
```diff
@@ -0,0 +1,296 @@
+
+Note that the OCI standard does not specify an `update` command.
+
+See issue #380 for more information.
```
missed link to issue

jodh-intel commented:
Hi @gorozco1 - gah! I had documented on this PR that the links should become active once it lands, but... you are right - they don't which is pretty sucky! ;-(

Changes:

- Made "internal" links active by specifying the full URL.
- Moved all "See issue ..." comments to the end of each section for consistency.
- Changed all `cc-oci-runtime` "See issue ..." comments to end with "... for more information." for consistency with the internal issue links.

grahamwhaley commented:
Nice @jodh-intel . Still
lgtm
but you hit the dreaded CRI-O semaphore failure....

grahamwhaley commented:
I could merge, but technically github would like @gorozco1 to OK the changes requested (broken links). let's wait for that ack...

jodh-intel commented:
Wiki updated to point to the in-tree doc:

- https://github.com/clearcontainers/runtime/wiki/Current-Limitations

(and the draft wiki doc this PR was based upon has been deleted: https://github.com/clearcontainers/runtime/wiki/Current-Limitations-2).
